### PR TITLE
Stop updating an invalid worksheet name in the frontend

### LINKED ISSUE #3404: No Error Message when changing worksheet name to an invalid one
**Describe the bug**
no error prompt when changing worksheet name to an invalid one.
Refresh the page and the name would remain the same.
ezgif com-gif-maker (2)

**Expected behavior**
An error message to be shown. The change should not succeed.

**Screenshots**

**Additional context**

### PR DESCRIPTION
**Pull request by yuqijin**
### Reasons for making this change

Fix the bug of showing no Error Message when changing worksheet name to an invalid one.
resolve a TODO in the code.

### Related issues

fixes #3404 

### Screenshots

ezgif com-gif-maker (5)

### Checklist

* [x] I've added a screenshot of the changes, if this is a frontend change
* [x] I've added and/or updated tests, if this is a backend change
* [x] I've run the [pre-commit.sh](https://github.com/codalab/codalab-worksheets/blob/master/pre-commit.sh) script
* [x] I've updated [docs](https://github.com/codalab/codalab-worksheets/tree/master/docs), if needed

### CODE REVIEW

File: frontend/src/components/EditableField.js [hunk 1]
```diff
@@ -73,8 +73,14 @@ class EditableFieldBase extends React.Component<{
                 if (this.props.onChange) this.props.onChange(this.state.value);
             })
             .fail(function(response, status, err) {
-                // TODO: this doesn't stop the value from updating in the frontend
-                console.log('Invalid value entered: ', response.responseText);
+                if (this.props.onError)
+                    this.props.onError('Invalid value entered: ' + response.responseText);
+                // Restore the original value
+                this.setState(() => {
```
Comment (epicfaace) [inline frontend/src/components/EditableField.js]: What's the need to add a function here? Could we just do `this.setState({value: this.props.value})`?

File: frontend/src/components/worksheets/Worksheet/WorksheetHeader.js [hunk 2]
```diff
@@ -78,6 +79,7 @@ export default ({
                                         value={info && info.name}
                                         uuid={info && info.uuid}
                                         onChange={() => reloadWorksheet()}
+                                        onError={(message) => openErrorMessageDialog(message)}
```
Comment (epicfaace) [inline frontend/src/components/worksheets/Worksheet/WorksheetHeader.js]: For simplicity
```suggestion
                                        onError={openErrorMessageDialog}
```

File: frontend/src/components/EditableField.js (hunk 1 above)
Comment (yuqijin) [reply frontend/src/components/EditableField.js]: you are right! Removed the function.

File: frontend/src/components/worksheets/Worksheet/Worksheet.js [hunk 3]
```diff
@@ -1954,6 +1961,7 @@ class Worksheet extends React.Component {
                     toggleCmdDialog={this.toggleCmdDialog}
                     toggleInformationModal={this.toggleInformationModal}
                     toggleCmdDialogNoEvent={this.toggleCmdDialogNoEvent}
+                    openErrorMessageDialog={this.openErrorMessageDialog}
```
Comment (epicfaace) [inline frontend/src/components/worksheets/Worksheet/Worksheet.js]: I think it may be clearer to rename this function to onError.

**Review by epicfaace — APPROVED** [review]

File: frontend/src/components/worksheets/Worksheet/Worksheet.js (hunk 3 above)
Comment (yuqijin) [reply frontend/src/components/worksheets/Worksheet/Worksheet.js]: updated

**Pull request merged**